src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt

```diff
@@ -36,41 +36,45 @@ class ObdDeviceConnection(private val inputStream: InputStream, private val outp
         command.handleResponse(obdRawResponse)
     }
 
-    private fun runCommand(command: ObdCommand, delayTime: Long): ObdRawResponse {
-        var rawData: String
+    private suspend fun runCommand(command: ObdCommand, delayTime: Long): ObdRawResponse {
+        var rawData = ""
         val elapsedTime = measureTimeMillis {
             sendCommand(command, delayTime)
             rawData = readRawData()
         }
         return ObdRawResponse(rawData, elapsedTime)
     }
 
-    private fun sendCommand(command: ObdCommand, delayTime: Long = 0) {
-        outputStream.write("${command.rawCommand}\r".toByteArray())
-        outputStream.flush()
-        if (delayTime > 0) {
-            sleep(delayTime)
+    private suspend fun sendCommand(command: ObdCommand, delayTime: Long = 0) = runBlocking {
+        withContext(Dispatchers.IO) {
+            outputStream.write("${command.rawCommand}\r".toByteArray())
+            outputStream.flush()
+            if (delayTime > 0) {
+                sleep(delayTime)
+            }
         }
     }
 
-    private fun readRawData(): String {
+    private suspend fun readRawData(): String = runBlocking {
         var b: Byte
         var c: Char
         val res = StringBuffer()
 
-        // read until '>' arrives OR end of stream reached (-1)
-        while (inputStream.available() > 0) {
-            b = inputStream.read().toByte()
-            if (b < 0) {
-                break
-            }
-            c = b.toInt().toChar()
-            if (c == '>') {
-                break
+        withContext(Dispatchers.IO) {
+            // read until '>' arrives OR end of stream reached (-1)
+            while (inputStream.available() > 0) {
+                b = inputStream.read().toByte()
+                if (b < 0) {
+                    break
+                }
+                c = b.toChar()
```

Comment:
Just one small thing 😅
Seems like in the newest kotlin version, we can't convert directly from bytes to char, having to convert to int first. Would you mind changing this line to this instead?
```suggestion
                c = b.toInt().toChar()
```